src/Npgsql/ConnectorPool.cs

```diff
@@ -260,7 +263,7 @@ internal async ValueTask<NpgsqlConnector> AllocateLong(NpgsqlConnection conn, Np
                     // Start the pruning timer if we're above MinPoolSize
                     if (_pruningTimer == null && newState.Total > _min)
                     {
-                        var newPruningTimer = new Timer(PruneIdleConnectors);
+                        var newPruningTimer = new Timer(s => PruneIdleConnectors(s));
```

Comment:
Any reason to do this? In that case you're creating a new method inside an internal class which internally calls `PruneIdleConnectors`.

Reply:
You're right, long day, it actually still captures the this reference. Thought it was a static method. Where what I did actually makes sense as C# doesn't cache any method group conversions but it does cache non capturing lambdas. 

Let me do a revision of that commit as I see a few other places like it to save allocations.